packages/extension/src/components/pages/ShareAddress/ShareAddress.tsx

```diff
@@ -49,12 +49,16 @@ export const ShareAddress: FC = () => {
     saveTrustedApp({ url: String(url), permissions }, selectedWallet);
     const currentWallet = getCurrentWallet();
     chrome.runtime
-      .sendMessage<ReceiveAddressBackgroundMessage>({
+      .sendMessage<ReceiveGetAddressBackgroundMessage>({
         app: GEM_WALLET,
-        type: 'RECEIVE_ADDRESS',
+        type: 'RECEIVE_GET_ADDRESS/V3',
         payload: {
           id,
-          publicAddress: currentWallet?.publicAddress
+          result: currentWallet?.publicAddress
```

Comment:
You don't have to wrap in a result object here, you can wrap only in the API - Just please keep consistency between all the pages.